src/components/atoms/InputField/InputField.tsx

```diff
@@ -61,9 +61,20 @@ const InputField: React.ForwardRefRenderFunction<
 
   const inputClassNames = classNames("input-field__input", inputClassName);
 
+  const keyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Escape" && e.target) {
+      e.currentTarget.blur();
+    }
+  };
+
   return (
     <div className={containerClassNames}>
-      <input ref={ref} className={inputClassNames} {...extraInputProps} />
+      <input
+        ref={ref}
+        className={inputClassNames}
+        onKeyDown={keyDown}
+        {...extraInputProps}
```

Comment:
If external code passes the `onKeyDown` prop, then this will end up being overwritten. To fix it we could either create a new `useCallback`'d 'wrapping function' that will call the passed in `onKeyDown` _and_ our handler defined in this code, or we could fix `useMousetrap` to properly use a local `ref`, which would then mean we shouldn't need to explicitly use the `onKeyDown` prop here.

Reply:
any help on this would be appreciated since sounds beyond my react-free comprehension ;) (attn @soichih)

Reply:
Fwiw, my preference would be for us to fix `useMousetrap` so it can bind to single elements using `ref`.

Reply:
Actually, seems I backlogged a thing about it on our internal issues: 🔒 https://github.com/sparkletown/internal-sparkle-issues/issues/376

Since you won't be able to see that:

> > We'll also want to ensure that the 'scope' of the keyboard shortcuts is limited to the right place.  For that, we'll need to fix the long outstanding bug in `useMousetrap`, that will cause some kind of error/issue/just not work when used with `withGlobalBind: false`:
> > 
> > - https://github.com/sparkletown/sparkle/blob/staging/src/hooks/useMousetrap.ts
> > 
> > _Originally posted by @0xdevalias in https://github.com/sparkletown/internal-sparkle-issues/issues/371#issuecomment-778909908_
> 
> _Originally posted by @0xdevalias in https://github.com/sparkletown/internal-sparkle-issues/issues/376_

Apparently I didn't include a whole lot of detail..

Reply:
> Fwiw, my preference would be for us to fix useMousetrap so it can bind to single elements using ref.

I would have loved that too but overall feels like OOS for the purpose of this PR since it is not the one introduced useMousetrap  and might take awhile, and there are bigger/nastier fishes to fry ATM.  As far as I see it - it all works ATM and there is that saying "ain't broken, don't fix it" ;)

Reply:
> and there is that saying "ain't broken, don't fix it"

Well, it is broken currently, as described in https://github.com/sparkletown/sparkle/pull/1416#discussion_r638466103

Fixing `useMousetrap` + using that is the 'optimal best' solution. The `useCallback` workaround is the 'passable for now' hack solution.

Reply:
Re

> If external code passes the onKeyDown prop,

Could it be checked at compile time and crash if so to provide safeguard for now?

Reply:
Even if we could, there's no point/it would be more convoluted/slower to do than to just implement the solution I described originally in https://github.com/sparkletown/sparkle/pull/1416#discussion_r638466103

Reply:
I have resolved merge conflicts and pushed -- it still works nicely.  If you see how it could be improved to address  shortcomings of used here existing sparkle codebase -- please do not hesitate to push them directly.  Even with your nice descriptions of desired changes they are "above my react-grade" (which is very very low ;-))